src/decoder/mod.rs

```diff
@@ -482,15 +482,17 @@ impl<R: BufRead + Seek> Reader<R> {
     /// Returns the next processed row of the image (discarding `InterlaceInfo`).
     ///
     /// See also [`Reader.read_row`], which reads into a caller-provided buffer.
-    pub fn next_row(&mut self) -> Result<Option<Row>, DecodingError> {
+    pub fn next_row<'a>(&'a mut self) -> Result<Option<Row<'a>>, DecodingError> {
```

Comment:
Nit: If these are with regards to (clippy) lints about hidden unexpected lifetimes, the more common option should be using the unnamed lifetime where possible.

```suggestion
    pub fn next_row(&mut self) -> Result<Option<Row<'_>>, DecodingError> {
```

There is a subtle difference in function signature with an explicit lifetime generic parameter. (And slightly more pleasant to read).

Reply:
Ok - done in all 4 cases.